webapp/components/survey/CategoryDetails/store/actions/category/useInit.js

```diff
@@ -0,0 +1,54 @@
+import { useCallback } from 'react'
+import { useDispatch } from 'react-redux'
+import axios from 'axios'
+
+import * as A from '@core/arena'
+import * as Category from '@core/survey/category'
+
+import { SurveyActions, useSurveyId } from '@webapp/store/survey'
+
+import { useFetchLevelItems } from '../item/useFetchLevelItems'
+import { State } from '../../state'
+
+const _createCategory = async ({ surveyId, onCategoryCreated, dispatch }) => {
+  const {
+    data: { category },
+  } = await axios.post(`/api/survey/${surveyId}/categories`, Category.newCategory())
+
+  if (onCategoryCreated) {
+    onCategoryCreated(category)
+  }
+  dispatch(SurveyActions.metaUpdated())
+
+  return category
+}
+
+const _fetchCategory = async ({ surveyId, categoryUuid }) => {
+  const {
+    data: { category },
+  } = await axios.get(`/api/survey/${surveyId}/categories/${categoryUuid}?draft=true&validate=true`)
+
+  return category
+}
```

Comment:
now this is on Service/api

Reply:
this file needs to be updated when the new ButtonMetaItemAdd will be used to create a new category... can you update this too in your pull request?